src/emitter.cpp

```diff
@@ -678,10 +682,19 @@ Emitter& Emitter::Write(const std::string& str) {
   if (!good())
     return *this;
 
-  const bool escapeNonAscii = m_pState->GetOutputCharset() == EscapeNonAscii;
+  StringEscaping::value stringEscaping = StringEscaping::None;
```

Comment:
Might make sense to put this as a "constructor" to the enum rather than a function here. Then each case of the switch statement could just be a "return".

Reply:
Done.